challenge_14/action_controller.rb

```diff
@@ -37,48 +76,10 @@ def call(environment)
 
   def draw_routes
     @router.draw do
-      get '/', to: -> environment {
-        response = Rack::Response.new
-        response.write '<p><strong>Submit a new Blog Post!</p></strong>'
-        response.write "<form method='post' enctype='application/x-www-form-urlencoded' action='/create-post'>"
-        response.write "<p><label>Blog Title: <input name='title'></label></p>"
-        response.write "<p><label>Content: <textarea name='content'></textarea></label></p>"
-        response.write '<p><button>Submit post</button></p>'
-        response.write '</form>'
-        response.content_type = 'text/html'
-        response.finish
-      }
-      get '/show-data', to: -> environment {
-        response = Rack::Response.new
-        response.content_type = 'text/html'
-        response.finish do
-          response.write '<ul>'
-    
-          Blog.all.each do |blog|
-            response.write '<li>'
-            response.write "<strong>Title: #{CGI.escape_html(blog.title)}</strong>, Content: #{CGI.escape_html(blog.content)}"
-            response.write '</li>'
-          end
-          response.write '</ul>'
-        end
-      }
-      post 'create-post', to: -> environment {
-        response = Rack::Response.new
-        request = Rack::Request.new(environment)
-        puts 'Got a new POST request!'
-    
-        Blog.create!(request.params)
-        response.redirect('/show-data', 303)
-        response.finish
-      }
-      match '*path', via: :all, to: -> environment {
-        response = Rack::Response.new
-        request = Rack::Request.new(environment)
-        response.write "Sorry, I don’t know what #{request.path_info} is"
-        response.content_type = 'text/plain'
-        response.status = 404
-        response.finish
-      }
+      get '/', to: AppController.action(:root)
```

Comment:
I should’ve mentioned this [last time](https://github.com/adrianna-chang-shopify/learning-project/pull/12), but `ActionDispatch::Routing::Mapper` has a [`#root` method](https://github.com/rails/rails/blob/v6.1.3.1/actionpack/lib/action_dispatch/routing/mapper.rb#L1680-L1698) which is [shorthand for](https://github.com/rails/rails/blob/v6.1.3.1/actionpack/lib/action_dispatch/routing/mapper.rb#L1989-L1992) `match '/', via: :get`, so you could use that here if you’d like.

Reply:
Oh cool, I've definitely seen (and used!) `root` in `routes.rb` files in Rails apps, but I never fully pieced together that that's shorthand for `match '/', via: :get`!